implementation.js

```diff
@@ -1,6 +1,78 @@
 'use strict';
 var ES = require('es-abstract/es6');
 var supportsDescriptors = require('define-properties').supportsDescriptors;
+var has = require('has');
+var global = require('system.global')();
+
+var parseIterable = function (iterator) {
+	var done = false;
+	var iterableResponse;
+	var tempArray = [];
+
+	if (iterator && typeof iterator.next === 'function') {
+		while (!done) {
+			iterableResponse = iterator.next();
+			if (
+				has(iterableResponse, 'value')
+				&& has(iterableResponse, 'done')
+			) {
+				if (iterableResponse.done === true) {
+					done = true;
+					break; // eslint-disable-line no-restricted-syntax
+
+				} else if (iterableResponse.done !== false) {
+					break; // eslint-disable-line no-restricted-syntax
+				}
+
+				tempArray.push(iterableResponse.value);
+			} else if (iterableResponse.done === true) {
+				done = true;
+				break; // eslint-disable-line no-restricted-syntax
+			} else {
+				break; // eslint-disable-line no-restricted-syntax
+			}
+		}
+	}
+
+	return done ? tempArray : false;
+};
+
+var hasSymbols = typeof Symbol === 'function' && typeof Symbol.iterator === 'symbol';
+var iteratorSymbol;
+if (hasSymbols) {
+	iteratorSymbol = Symbol.iterator;
+} else if ('Set' in global) {
+	var supportsStrIterator = (function () {
+		try {
+			Function( // eslint-disable-line no-new-func
+				'for (var x of new Set()) {}'
+			)();
+			return true;
+		} catch (e) {
+			return false;
+		}
+	}());
+
+	if (supportsStrIterator) {
+		iteratorSymbol = '@@iterator';
+	}
+}
+
+var isCallable = require('is-callable');
+
+var usingIterator = function (items) {
+	if (iteratorSymbol && isCallable(items[iteratorSymbol])) {
+		return items[iteratorSymbol]();
+	} else if ('Set' in global && 'Map' in global && isCallable(items.entries) && isCallable(items.values)) {
+		if (items.constructor === Set) {
```

Comment:
I made a JSBin to test this out, it seems `Set`s and `Map`s are detectable across realms (I tested web workers) via `instanceof`. Is this what you expected?

http://jsbin.com/kiguqunoxi/edit?js,console